### PR TITLE
[webview_flutter] Support new onUrlChanged event on Android.

### PR DESCRIPTION
**Pull request by BeMacized**
This PR adds Android support for the new onUrlChanged event added to the platform interface in #4509.

As this PR contains and depends on the changes made to the platform interface in #4509, this PR will remain in draft state until that PR has been successfully merged and the changes have been published.

Relevant issue: 
- flutter/flutter#27729

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/webview_flutter/webview_flutter_android/example/android/build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.3.0'
+        classpath 'com.android.tools.build:gradle:4.1.3'
```
Comment (mvanbeusekom) [inline packages/webview_flutter/webview_flutter_android/example/android/build.gradle:8]: is this change necessary to support the `onUrlChanged` feature?

The gradle version should be updated at some point to be in sync with the current Flutter stable (which uses version 4.1.0 at the moment). However it would be better to do so in a separate PR.

File: packages/webview_flutter/webview_flutter_android/example/android/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Fri Jun 23 08:50:38 CEST 2017
+#Mon Nov 15 16:20:35 CET 2021
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-bin.zip
```
Comment (mvanbeusekom) [inline packages/webview_flutter/webview_flutter_android/example/android/gradle/wrapper/gradle-wrapper.properties:6]: Same as with the version change of the `com.android.tools.build:gradle` plugin.

File: packages/webview_flutter/webview_flutter_android/example/android/build.gradle (hunk 1 above)
Comment (BeMacized) [reply packages/webview_flutter/webview_flutter_android/example/android/build.gradle]: I had not intended to update gradle, I think I just blindly accepted one of android studio's suggestions. Reverted it for now.

**Comment by Hixie** [comment]
@BeMacized are you still interested in working on this PR? Is there anything you need from anyone?

**Comment by BeMacized** [comment]
@Hixie We stopped working on this feature due to there being no good way to implement it in the current structure without a breaking change. Once [#94051](https://github.com/flutter/flutter/issues/94051) has been resolved I'll be happy to pick it back up again.

**Comment by Hixie** [comment]
@BeMacized should we close this PR in the meantime? We can always refer to the changes in a closed PR. (I ask just because this is one of our oldest PRs and I'm generally trying to clean up our PR queue.)

**Comment by BeMacized** [comment]
> @BeMacized should we close this PR in the meantime? We can always refer to the changes in a closed PR. (I ask just because this is one of our oldest PRs and I'm generally trying to clean up our PR queue.)

Yes, I'm closing this. I'll probably be rewriting this feature once 4.0 has been released.
